sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -475,116 +469,110 @@ public EncryptedBlobAsyncClient getCustomerProvidedKeyAsyncClient(CustomerProvid
+
+                        int encryptedBytes;
+                        try {
+                            encryptedBytes = cipher.update(plainTextBuffer, encryptedTextBuffer);
+                        } catch (ShortBufferException e) {
+                            throw logger.logExceptionAsError(Exceptions.propagate(e));
+                        }
+                        encryptedTextBuffer.position(0);
+                        encryptedTextBuffer.limit(encryptedBytes);
+                        return encryptedTextBuffer;
+                    });
 
                     /*
-                    Defer() ensures the contained code is not executed until the Flux is subscribed to, in
-                    other words, cipher.doFinal() will not be called until the plainTextFlux has completed
-                    and therefore all other data has been encrypted.
+                     * Defer() ensures the contained code is not executed until the Flux is subscribed to, in
+                     * other words, cipher.doFinal() will not be called until the plainTextFlux has completed
+                     * and therefore all other data has been encrypted.
                      */
-                        encryptedTextFlux = Flux.concat(encryptedTextFlux, Flux.defer(() -> {
-                            try {
-                                return Flux.just(ByteBuffer.wrap(cipher.doFinal()));
-                            } catch (GeneralSecurityException e) {
-                                throw logger.logExceptionAsError(Exceptions.propagate(e));
-                            }
-                        }));
-                        return new EncryptedBlob(encryptionData, encryptedTextFlux);
-                    }));
-        } catch (NoSuchAlgorithmException | NoSuchPaddingException e) {
-            // These are hardcoded and guaranteed to work. There is no reason to propogate a checked exception.
+                    encryptedTextFlux = Flux.concat(encryptedTextFlux,
+                        Mono.fromCallable(() -> ByteBuffer.wrap(cipher.doFinal())));
+                    return new EncryptedBlob(encryptionData, encryptedTextFlux);
+                }));
+        } catch (GeneralSecurityException e) {
+            // These are hardcoded and guaranteed to work. There is no reason to propagate a checked exception.
             throw logger.logExceptionAsError(new RuntimeException(e));
         }
     }
 
+    SecretKey generateSecretKey() throws NoSuchAlgorithmException {
+        KeyGenerator keyGen = KeyGenerator.getInstance(CryptographyConstants.AES);
+        keyGen.init(256);
+
+        return keyGen.generateKey();
+    }
+
+    Cipher generateCipher(SecretKey aesKey) throws GeneralSecurityException {
+        Cipher cipher = Cipher.getInstance(CryptographyConstants.AES_CBC_PKCS5PADDING);
+
+        // Generate content encryption key
+        cipher.init(Cipher.ENCRYPT_MODE, aesKey);
+
+        return cipher;
+    }
```

Comment:
I'm guessing this is to make these overridable and have "spy" in tests. Should we take KeyGenerator and Cipher references in ctor instead?